src/libraries/System.Private.CoreLib/src/System/IO/File.cs

```diff
@@ -689,6 +689,33 @@ public static void WriteAllBytes(string path, byte[] bytes)
             RandomAccess.WriteAtOffset(sfh, bytes, 0);
         }
 
+        public static void AppendAllBytes(string path, byte[] bytes)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(path);
+            ArgumentNullException.ThrowIfNull(bytes);
+
+            using SafeFileHandle fileHandle = OpenHandle(path, FileMode.Open, FileAccess.Write, FileShare.Read);
```

Comment:
In other places in the library where we append, we're using [`FileMode.Append`](https://learn.microsoft.com/en-us/dotnet/api/system.io.filemode?view=net-7.0) and it takes care of the seek-to-end... would it be better to use that FileMode here?

Reply:
I think you miss the details.

for example, this is the "AppendAllText"

```
        public static void AppendAllText(string path, string? contents, Encoding encoding)
        {
            Validate(path, encoding);

            WriteToFile(path, FileMode.Append, contents, encoding);
        }
```

which goes to "WriteToFile" :

```
        private static void WriteToFile(string path, FileMode mode, string? contents, Encoding encoding)
        {
            ReadOnlySpan<byte> preamble = encoding.GetPreamble();
            int preambleSize = preamble.Length;

            using SafeFileHandle fileHandle = OpenHandle(path, mode, FileAccess.Write, FileShare.Read, FileOptions.None, GetPreallocationSize(mode, contents, encoding, preambleSize));
            long fileOffset = mode == FileMode.Append && fileHandle.CanSeek ? RandomAccess.GetLength(fileHandle) : 0;

            if (string.IsNullOrEmpty(contents))
            {
                if (preambleSize > 0 // even if the content is empty, we want to store the preamble
                    && fileOffset == 0) // if we're appending to a file that already has data, don't write the preamble.
                {
                    RandomAccess.WriteAtOffset(fileHandle, preamble, fileOffset);
                }
                return;
            }
```

Reply:
I agree. I changed the FileMode to Append.

Reply:
I think in this case, given that you are using `RandomAcesss`, using Append may be worse since you are telling the handle to seek to the end-of-file and then you proceed to use `RandomAccess.WriteAtOffset` which doesn't care about the seek.

@adamsitnik thoughts?